### PR TITLE
Add ISO Macro keyboard

### PR DESCRIPTION
**Pull request by duoshock**
## Description

ISO Macro is a macro pad with ISO enter key and encoders.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/merge/iso_macro/config.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_H
+#define CONFIG_H
```
Comment (drashna) [inline keyboards/merge/iso_macro/config.h:19]:
```suggestion
#pragma once
```

File: keyboards/merge/iso_macro/config.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+#include "config_common.h"
+
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x1200
+#define DEVICE_VER 		0x0001
```
Comment (drashna) [inline keyboards/merge/iso_macro/config.h:25]:
```suggestion
#define DEVICE_VER      0x0001
```

File: keyboards/merge/iso_macro/iso_macro.c [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#include "iso_macro.h"
```
Comment (drashna) [inline keyboards/merge/iso_macro/iso_macro.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/merge/iso_macro/iso_macro.c [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+
+void matrix_init_kb(void) {
+	matrix_init_user();
+}
```
Comment (drashna) [inline keyboards/merge/iso_macro/iso_macro.c:5]: This should be removed.
```suggetion
```

File: keyboards/merge/iso_macro/iso_macro.h [hunk 5]
```diff
@@ -0,0 +1,16 @@
+#ifndef ISOMACRO_H
+#define ISOMACRO_H
```
Comment (drashna) [inline keyboards/merge/iso_macro/iso_macro.h:2]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

Also: 
```suggestion
#pragma once
```

File: keyboards/merge/iso_macro/iso_macro.h [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+      k21, k00, k01, k02, k20, \
+      k22, k10, k11, k12 \
+) \
+{ \
+    { k00, k01, k02 }, \
+    { k10, k11, k12 }, \
+    { k20, k21, k22 }  \
+} 
+
+#endif
```
Comment (drashna) [inline keyboards/merge/iso_macro/iso_macro.h:16]:
```suggestion
```

File: keyboards/merge/iso_macro/config.h [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+
+#define ENCODERS_PAD_A { F0, D0 }
+#define ENCODERS_PAD_B { F1, D1 }
+#define ENCODER_RESOLUTION 4
+
+
+#endif
```
Comment (drashna) [inline keyboards/merge/iso_macro/config.h:48]:
```suggestion
```

File: keyboards/merge/iso_macro/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,49 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/merge/iso_macro/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/merge/iso_macro/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,49 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/merge/iso_macro/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/merge/iso_macro/rules.mk [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+NKRO_ENABLE = no            # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+API_SYSEX_ENABLE = no
+TAP_DANCE_ENABLE = no
+ENCODER_ENABLE = yes
```
Comment (drashna) [inline keyboards/merge/iso_macro/rules.mk:32]:
```suggestion
AUDIO_ENABLE = no           # Audio output
BLUETOOTH_ENABLE = no       # Enable Bluetooth
RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
ENCODER_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/merge/iso_macro/config.h (hunk 1 above)
Comment (duoshock) [reply keyboards/merge/iso_macro/config.h:19]: Thanks for the feedback. I've changed all requests accordingly.

**Comment by duoshock** [comment]
Thanks for the feedback. I've changed all the request accordingly.

File: keyboards/merge/iso_macro/config.h [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+#pragma once
+
+#include "config_common.h"
+
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x1200
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Merge
+#define PRODUCT         ISO Macro
+#define DESCRIPTION     ISO Macro
```
Comment (fauxpark) [inline keyboards/merge/iso_macro/config.h:26]:
```suggestion
```

File: keyboards/merge/iso_macro/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+[2] = LAYOUT( \
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+),
+
+[3] = LAYOUT( \
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+)
+
```
Comment (fauxpark) [inline keyboards/merge/iso_macro/keymaps/default/keymap.c:47]:
```suggestion
[0] = LAYOUT(
  KC_HOME, KC_4, KC_5, KC_6, KC_ENT,
  KC_MUTE, KC_1, KC_2, KC_3
),
```

File: keyboards/merge/iso_macro/keymaps/via/iso_macro.json [hunk 13]
```diff
@@ -0,0 +1,62 @@
+{
```
Comment (fauxpark) [inline keyboards/merge/iso_macro/keymaps/via/iso_macro.json:1]: This file goes in the VIA repo, not here.

File: keyboards/merge/iso_macro/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+),
+
+[2] = LAYOUT( \
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+),
+
+[3] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/merge/iso_macro/keymaps/via/keymap.c:43]:
```suggestion
[0] = LAYOUT(
  BL_BRTG, KC_4, KC_5, KC_6, KC_ENT,
  KC_MUTE, KC_1, KC_2, KC_3
),

[1] = LAYOUT(
  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
),

[2] = LAYOUT(
  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
),

[3] = LAYOUT(
```

File: keyboards/merge/iso_macro/readme.md [hunk 15]
```diff
@@ -0,0 +1,14 @@
+# ISO Macro
+
+![ISO Macro](https://i.imgur.com/wJoaZ8x.jpg)
```
Comment (fauxpark) [inline keyboards/merge/iso_macro/readme.md:3]:
```suggestion
![ISO Macro](https://i.imgur.com/wJoaZ8xl.jpg)
```

File: keyboards/merge/iso_macro/rules.mk [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/merge/iso_macro/rules.mk:11]:
```suggestion
```

File: keyboards/merge/iso_macro/rules.mk [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+NKRO_ENABLE = no            # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+ENCODER_ENABLE = yes
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/merge/iso_macro/rules.mk:32]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by duoshock** [comment]
Thanks. I've updated it.

File: keyboards/merge/iso_macro/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+/* LAYER 0
+ * ,---------------------------------------------.
+ * | Encoder A  |   4   |   5   |   6   | Enter  |
+ * |------------|-------+-------+-------|        |
+ * | Encoder B  |   1   |   2   |   3   |        |
+ * `---------------------------------------------'
+ */
+[0] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/merge/iso_macro/keymaps/default/keymap.c:28]:
```suggestion
[0] = LAYOUT(
```

File: keyboards/merge/iso_macro/keymaps/default/readme.md [hunk 19]
```diff
@@ -0,0 +1,5 @@
+![ISO Macro Layout Image](https://i.imgur.com/5NEfKVz.jpg)
+
+# Default ISO Macro Layout
```
Comment (fauxpark) [inline keyboards/merge/iso_macro/keymaps/default/readme.md:3]:
```suggestion
# Default ISO Macro Layout

![ISO Macro Layout Image](https://i.imgur.com/5NEfKVz.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by duoshock** [comment]
It's updated. Thanks!

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**